### DISCUSSION TITLE
Allow author to approve own PR if someone else has committed

### DISCUSSION THREAD

**Discussion by quilicicf** [community/community#120835] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

A common use case I regularly stumble upon is a PR that starts with one author (A) and then gets finished by another (B).

Let's say A starts the development, then falls sick or goes on holidays and B finishes the work. 
Now, A cannot approve the PR even though he's not the last person to have worked on it (and no longer the main developer).

I'm wondering how to fix this and I see two potential solutions but I'm not sure I understand all the consequences (in terms of security) so thoughts on this would be appreciated:
* One solution would be to allow (via a setting?) the author to review their own PR if someone else has committed
* Another one would be to allow changing the author (or having another role?) so that B takes over and A becomes a reviewer instead

Note: [there's another discussion](https://github.com/orgs/community/discussions/23134) that points that B can commit to the PR, approve and merge and A now has their name on a commit that doesn't do what he developed (security issue?). I think the two issues are linked but the proposed solution that there should be a third reviewer to approve doesn't quite cut it as the team can get short on reviewers and this is blocking where some teams are mature enough (or codebases not sensitive enough...) to require more flexibility.

**Comment by benjaminclark177-lab** [comment]
A practical approach might be integrating [LemLedger ](https://lemledger.com/)to track contributions accurately, so the author and reviewer roles stay clear without compromising security.

**Reply by quilicicf** [reply]
I don't see how an external tool could help getting GitHub to change its behavior though? :thinking:
